store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/schema/FederatedViewValidator.java

```diff
@@ -56,14 +61,38 @@ public ValidationResult validate(final View view, final Schema schema, final Set
     }
 
     protected ValidationResult getEdgeResult(final View view, final Schema schema, final Set<StoreTrait> storeTraits, final boolean isStoreOrdered) {
-        final ValidationResult edgeResult = new ValidationResult();
-        validateEdge(view, schema, storeTraits, isStoreOrdered, edgeResult);
-        return edgeResult;
+        final Function<Map.Entry<String, ViewElementDefinition>, ValidationResult> validationResultFunction = e -> {
+            final ValidationResult singleResult = new ValidationResult();
+            validateEdge(new View.Builder().edge(e.getKey(), e.getValue()).build(), schema, storeTraits, isStoreOrdered, singleResult);
+            return singleResult;
+        };
+
+        return getElementResult(validationResultFunction, view.getEdges());
     }
 
     protected ValidationResult getEntityResult(final View view, final Schema schema, final Set<StoreTrait> storeTraits, final boolean isStoreOrdered) {
-        final ValidationResult entitiesResult = new ValidationResult();
-        validateEntities(view, schema, storeTraits, isStoreOrdered, entitiesResult);
-        return entitiesResult;
+
+        final Function<Map.Entry<String, ViewElementDefinition>, ValidationResult> validationResultFunction = e -> {
+            final ValidationResult singleResult = new ValidationResult();
+            validateEntities(new View.Builder().entity(e.getKey(), e.getValue()).build(), schema, storeTraits, isStoreOrdered, singleResult);
+            return singleResult;
+        };
+        return getElementResult(validationResultFunction, view.getEntities());
+    }
+
+    private ValidationResult getElementResult(final Function<Map.Entry<String, ViewElementDefinition>, ValidationResult> validationResultFunction, final Map<String, ViewElementDefinition> elements) {
+        final ValidationResult rtn = new ValidationResult();
+
+        final List<ValidationResult> viewResults = elements.entrySet().stream()
+                .map(validationResultFunction)
+                .collect(Collectors.toList());
```

Comment:
Couldn't you just add it to the validation result here?

Reply:
No I don't want the errors if one is successful. So I got to check first, if all fail, then I add them to the return